sign.sh

```diff
@@ -132,11 +136,49 @@ signRelease()
           file=$(basename "$f")
           mv "$f" "${dir}/unsigned_${file}"
           curl --fail --silent --show-error -o "$f" -F file="@${dir}/unsigned_${file}" -F entitlements="@$ENTITLEMENTS" https://cbi.eclipse.org/macos/codesign/sign
-          chmod --reference="${dir}/unsigned_${file}" "$f"
-          rm -rf "${dir}/unsigned_${file}"
+          echo File = "$f"
+          TESTMACSIGN=$(grep -ic "$MACSIGNSTRING" "$f")
+          echo Sign Result = "$TESTMACSIGN"
+          if [ "$TESTMACSIGN" -gt 0 ]
+          then
+            echo "Code Signed For File $f"
+            chmod --reference="${dir}/unsigned_${file}" "$f"
+            rm -rf "${dir}/unsigned_${file}"
+          else
+            max_iterations=20
+            iteration=1
+            success=false
+            errcount=0
+            echo "Code Not Signed For File $f"
+            while [ $iteration -le $max_iterations ] && [ $success = false ]; do
+              echo $iteration Of $max_iterations
+              sleep 1
+              curl --fail -o "$f" -F file="@${dir}/unsigned_${file}" -F entitlements="@$ENTITLEMENTS" https://cbi.eclipse.org/macos/codesign/sign
+              TESTMACSIGN2=$(grep -ic "$MACSIGNSTRING" "$f")
+              echo TESTMACSIGN2 = "$TESTMACSIGN2"
+              if [ "$TESTMACSIGN2" -gt 0 ]
+              then
+                echo "$f Signed OK On Attempt $iteration"
+                chmod --reference="${dir}/unsigned_${file}" "$f"
+                rm -rf "${dir}/unsigned_${file}"
+                success=true
+              else
+                echo "$f Failed Signing On Attempt $iteration"
+                success=false
+                iteration=$((iteration+1))
+                errcount=$((errcount+1))
+              fi
+            done
+            if [ "$errcount" -gt 0 ]
+            then
+              echo "Errors Encountered During Signing"
+              echo "Error Count = $errcount"
+              exit 1
```

Comment:
I'm not sure I understand. I thought this fix was intended to retry 20 times in order to *avoid* failures. This appears to say "If we needed to retry downloading this file even once, then fail".

How would this be better than the current situation?

Reply:
The multi level nested loops and if statements make this slightly difficult to read.

Reply:
Looks like its working ok in a successful run, Ive also tested it in my local test script, and it works correctly,, with a failure followed by a success

Reply:
Thanks Scott. Looks excellent, well done. :)